tests/test_bgp_af.rb

```diff
@@ -118,114 +157,151 @@ def test_default_information_originate
     vrf = 'red'
     af = %w(ipv4 unicast)
 
-    #
-    # Set and verify
-    #
-    bgp_af = RouterBgpAF.new(asn, vrf, af)
-    bgp_af.default_information_originate = true
-    assert(bgp_af.default_information_originate,
-           'Error: default-information originate not set')
+    if platform == :ios_xr
 
-    pattern = /^ *default-information originate$/
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+      # XR does not support default-information under address-family
+      config_ios_xr_dependencies(asn)
+      bgp_af = RouterBgpAF.new(asn, vrf, af)
 
-    assert_match(pattern, af_string,
-                 "Error: 'default_information originate' is not" \
-                   ' configured and should be')
+      assert_raises(Cisco::UnsupportedError) do
+        bgp_af.default_information_originate = true
+      end
+    elsif platform == :nexus
 
-    #
-    # Unset and verify
-    #
+      # Check default value
+      bgp_af = RouterBgpAF.new(asn, vrf, af)
+      initial = bgp_af.default_information_originate
+      default = bgp_af.default_default_information_originate
+      assert_equal(default, initial)
 
-    # Do a 'no default-information originate'
-    bgp_af.default_information_originate = false
+      # Toggle the state a few times
+      bgp_af.default_information_originate = !initial
+      assert_equal(!initial, bgp_af.default_information_originate)
 
-    pattern = /^ *default-information originate$/
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+      bgp_af.default_information_originate = initial
+      assert_equal(initial, bgp_af.default_information_originate)
 
-    refute_match(pattern, af_string,
-                 "Error: 'default_information originate' " \
-                   'is configured and should not be')
+      bgp_af.destroy
+    end
   end
 
   ##
-  ## client-to-client reflection
+  ## Negative tests with non-default vrf on ios_xr
   ##
-  def test_client_to_client
+  def client_to_client_vrf_xr_neg_test
     asn = '55'
     vrf = 'red'
     af = %w(ipv4 unicast)
-
     bgp_af = RouterBgpAF.new(asn, vrf, af)
-    pattern = /^ *client-to-client reflection$/
-    #
-    # Default is 'client-to-client' is configured
-    #
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    assert_nil(bgp_af.client_to_client,
+               'client_to_client should return nil on XR with non-default vrf')
+    assert_raises(Cisco::UnsupportedError) do
+      bgp_af.client_to_client = true
+    end
+  end
 
-    assert_match(pattern, af_string,
-                 "Error: 'client-to-client reflection' is not configured " \
-                   'and should be')
+  ##
+  ## client-to-client reflection (BT: trying a different style)
+  ##
+  def test_client_to_client
+    asn = '55'
```

Comment:
Note that l2vpn is perfectly valid address-family:

router bgp 55
 address-family l2vpn evpn

You should add these tests throughout. You can see bgp neighbor af for examples on how to do this within a matrix that can be called for each test method. It's a harness I think should be adopted.

Ditto ipv6 and multicast.